### ISSUE TITLE
invalid SQL when reading gpkg using bbox with pyogrio

### ISSUE THREAD

**Issue by Ganoash** [geopandas/pyogrio#179]
```python
from geopandas import read_file
from rasterio.coords import BoundingBox
from shapely.affinity import rotate
from shapely.geometry import box

def weird_error():
    # Use a breakpoint in the code line below to debug your script.
    geowindow = BoundingBox(left=97424.94802231214, bottom=458550.9043375151, right=97680.94802231214, top=458806.9043375151)

    rotated_window = rotate(box(*geowindow), 10)
    dataframe = read_file("./data_set.gpkg", bbox=rotated_window, engine="pyogrio", sql_dialect="SQLITE")
    return dataframe

# See PyCharm help at https://www.jetbrains.com/help/pycharm/
if __name__ == '__main__':
    weird_error()
```
[data_set.zip](https://github.com/geopandas/pyogrio/files/10168391/data_set.zip)

```
[home_dir]/.var/app/com.jetbrains.PyCharm-Professional/data/virtualenvs/error-wT7yEB3n/bin/python [home_dir]/Documents/codebase/error/main.py 
ERROR 1: sqlite3_prepare_v2(SELECT COUNT(*) FROM "rtree_mapped_reference_bgt_oot_2018_04_01_geom" WHERE maxx >= 97404.665663185180 AND minx <= 97701.230381439076 AND maxy >= 458530.621978388168 AND miny <= 458827.186696641962) failed: no such module: rtree
[home_dir]/.var/app/com.jetbrains.PyCharm-Professional/data/virtualenvs/error-wT7yEB3n/lib/python3.10/site-packages/pyogrio/raw.py:120: UserWarning: Layer 'b'mapped_reference_bgt_oot_2018_04_01'' does not have any features to read
  result = ogr_read(
ERROR 1: failed to prepare SQL: SELECT m."fid", m."geom", m."main_object_type", m."full_object_type", m."LV-publicatiedatum", m."bgt-functie", m."bgt-fysiekVoorkomen", m."bgt-status", m."bgt-type", m."bronhouder", m."eindRegistratie", m."gml_id", m."hoortBijTypeOverbrugging", m."identificatie.lokaalID", m."identificatie.namespace", m."identificatieBAGPND", m."inOnderzoek", m."nummeraanduidingreeks_1.identificatieBAGVBOHoogsteHuisnummer", m."nummeraanduidingreeks_1.identificatieBAGVBOLaagsteHuisnummer", m."nummeraanduidingreeks_1.positie_1.hoek", m."nummeraanduidingreeks_1.tekst", m."nummeraanduidingreeks_2.identificatieBAGVBOHoogsteHuisnummer", m."nummeraanduidingreeks_2.identificatieBAGVBOLaagsteHuisnummer", m."nummeraanduidingreeks_2.positie_1.hoek", m."nummeraanduidingreeks_2.tekst", m."nummeraanduidingreeks_3.identificatieBAGVBOHoogsteHuisnummer", m."nummeraanduidingreeks_3.identificatieBAGVBOLaagsteHuisnummer", m."nummeraanduidingreeks_3.positie_1.hoek", m."nummeraanduidingreeks_3.tekst", m."objectBeginTijd", m."objectEindTijd", m."opTalud", m."overbruggingIsBeweegbaar", m."plus-functie", m."plus-fysiekVoorkomen", m."plus-type", m."relatieveHoogteligging", m."tijdstipRegistratie", m."typeOverbruggingsdeel", m."full_object_type_unmapped" FROM "mapped_reference_bgt_oot_2018_04_01" m JOIN "rtree_mapped_reference_bgt_oot_2018_04_01_geom" r ON m."fid" = r.id WHERE r.maxx >= 97404.665663185180 AND r.minx <= 97701.230381439076 AND r.maxy >= 458530.621978388168 AND r.miny <= 458827.186696641962
Traceback (most recent call last):
  File "pyogrio/_io.pyx", line 647, in pyogrio._io.get_features
  File "pyogrio/_err.pyx", line 177, in pyogrio._err.exc_wrap_pointer
pyogrio._err.CPLE_AppDefinedError: failed to prepare SQL: SELECT m."fid", m."geom", m."main_object_type", m."full_object_type", m."LV-publicatiedatum", m."bgt-functie", m."bgt-fysiekVoorkomen", m."bgt-status", m."bgt-type", m."bronhouder", m."eindRegistratie", m."gml_id", m."hoortBijTypeOverbrugging", m."identificatie.lokaalID", m."identificatie.namespace", m."identificatieBAGPND", m."inOnderzoek", m."nummeraanduidingreeks_1.identificatieBAGVBOHoogsteHuisnummer", m."nummeraanduidingreeks_1.identificatieBAGVBOLaagsteHuisnummer", m."nummeraanduidingreeks_1.positie_1.hoek", m."nummeraanduidingreeks_1.tekst", m."nummeraanduidingreeks_2.identificatieBAGVBOHoogsteHuisnummer", m."nummeraanduidingreeks_2.identificatieBAGVBOLaagsteHuisnummer", m."nummeraanduidingreeks_2.positie_1.hoek", m."nummeraanduidingreeks_2.tekst", m."nummeraanduidingreeks_3.identificatieBAGVBOHoogsteHuisnummer", m."nummeraanduidingreeks_3.identificatieBAGVBOLaagsteHuisnummer", m."nummeraanduidingreeks_3.positie_1.hoek", m."nummeraanduidingreeks_3.tekst", m."objectBeginTijd", m."objectEindTijd", m."opTalud", m."overbruggingIsBeweegbaar", m."plus-functie", m."plus-fysiekVoorkomen", m."plus-type", m."relatieveHoogteligging", m."tijdstipRegistratie", m."typeOverbruggingsdeel", m."full_object_type_unmapped" FROM "mapped_reference_bgt_oot_2018_04_01" m JOIN "rtree_mapped_reference_bgt_oot_2018_04_01_geom" r ON m."fid" = r.id WHERE r.maxx >= 97404.665663185180 AND r.minx <= 97701.230381439076 AND r.maxy >= 458530.621978388168 AND r.miny <= 458827.186696641962

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "[home_dir]/Documents/codebase/error/main.py", line 17, in <module>
    weird_error()
  File "[home_dir]/Documents/codebase/error/main.py", line 11, in weird_error
    dataframe = read_file("./mapped_reference_bgt_oot_2018_04_01.gpkg", bbox=rotated_window, engine="pyogrio", sql_dialect="SQLITE")
  File "[home_dir]/.var/app/com.jetbrains.PyCharm-Professional/data/virtualenvs/error-wT7yEB3n/lib/python3.10/site-packages/geopandas/io/file.py", line 263, in _read_file
    return _read_file_pyogrio(
  File "[home_dir]/.var/app/com.jetbrains.PyCharm-Professional/data/virtualenvs/error-wT7yEB3n/lib/python3.10/site-packages/geopandas/io/file.py", line 408, in _read_file_pyogrio
    return pyogrio.read_dataframe(path_or_bytes, bbox=bbox, **kwargs)
  File "[home_dir]/.var/app/com.jetbrains.PyCharm-Professional/data/virtualenvs/error-wT7yEB3n/lib/python3.10/site-packages/pyogrio/geopandas.py", line 137, in read_dataframe
    meta, index, geometry, field_data = read(
  File " [home_dir]/.var/app/com.jetbrains.PyCharm-Professional/data/virtualenvs/error-wT7yEB3n/lib/python3.10/site-packages/pyogrio/raw.py", line 120, in read
    result = ogr_read(
  File "pyogrio/_io.pyx", line 925, in pyogrio._io.ogr_read
  File "pyogrio/_io.pyx", line 655, in pyogrio._io.get_features
ValueError: Invalid SQL query
```

With the above code:
Goal: read the gpkg information into a dataframe
Expected behaviour: the dataframe is read in filtered by the bounding box
Actual behaviour: An error occurs, possibly due to rtree in python.

The standard sqllite distribution in python does not include rtree support. Is there some way around this? And is it desirable to lean on functionality that is very hard to change in the standard library, or is there some other issue at play here. Some help would be appreciated

**Comment by jorisvandenbossche** [comment]
@Ganoash thanks for the report! Similarly as in https://github.com/geopandas/pyogrio/issues/187, this is probably caused because of the released wheels currently not being built with sqlite with rtree, but only sqlite without rtree. 

This _should_ be solved in the development version, and we have to make a release to get the new wheels out.

Would you like to test this? You can install the latest wheel by going to https://github.com/geopandas/pyogrio/actions/runs/3793070654, and at the bottom of that page download the "artifact". Unpack this archive, and then `pip install path/to/archive/pyogrio-0.4.2+12.g40e7659-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl` (or the correct wheel name for your OS)

**Comment by brendan-ward** [comment]
@Ganoash I believe this is resolved now as of 0.5.0 (I cannot reproduce using latest 0.5.1 wheels on macOS), so I'm closing this.  If this still does not work for you as of >= 0.5.0, please reopen this issue.
